packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_services.dart

```diff
@@ -118,15 +137,73 @@ class DeepLinksServices {
     );
   }
 
-  Future<Map<String, List<DomainError>>> validateIosDomain({
+  Future<ValidateIosDomainResult> validateIosDomain({
+    required String bundleId,
+    required String teamId,
     required List<String> domains,
   }) async {
-    //TODO(hangyujin): This is just a fake response, implement the iOS domain validation when the API is available.
     final domainErrors = <String, List<DomainError>>{
-      for (final domain in domains)
-        domain: <DomainError>[IosDomainError.existence],
+      for (final domain in domains) domain: <DomainError>[],
     };
-    return domainErrors;
+    //TODO(hangyujin): Add error code to the result.
+    const errorCode = '';
+
+    final domainsBybatch = _splitDomains(domains);
+
+    for (final domainList in domainsBybatch) {
+      final response = await http.post(
+        Uri.parse(_iosDomainValidationURL),
+        headers: postHeader,
+        body: jsonEncode({
+          _appIdKey: {
+            _bundleIdKey: bundleId,
+            _teamIdKey: teamId,
+          },
+          _universalLinkDomainsKey: [
+            for (final domain in domainList) {_iosDomainNameKey: domain},
+          ],
+        }),
+      );
+      final result = json.decode(response.body) as Map<String, Object?>;
+      final validationResult = (result[_iosValidationResultsKey] as List)
+          .cast<Map<String, Object?>>();
+
+      for (final domainResult in validationResult) {
+        final domainName = domainResult[_domainNameKey] as String?;
+        if (domainName == null) {
+          continue;
+        }
+        final failedChecks = (domainResult[_failedChecksKey] as List?)
+            ?.cast<Map<String, Object?>>();
+        if (failedChecks != null) {
```

Comment:
This can be consolidated to:
```dart
if (domainResult[_domainNameKey] case final String domainName) {
  if (domainResult[_failedChecksKey] case final List<Map<String, Object?>> failedChecks) {
    ...
  }
}
```

Not completely positive on the second if statement due to the casting, but you can give it a shot.

Reply:
Update the first part. 
Can't cast the second part like that so i kept it.